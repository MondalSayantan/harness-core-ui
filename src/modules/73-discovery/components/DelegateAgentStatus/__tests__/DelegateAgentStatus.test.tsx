/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import type { DatabaseDelegateTaskStatus } from 'services/servicediscovery'
import { DiscoveryAgentStatus } from '../DelegateAgentStatus'

describe('Discovery Agent Status ', () => {
  test('render', async () => {
    const props = {
      status: 'SUCCESS' as DatabaseDelegateTaskStatus | undefined
    }
    const { container } = render(<DiscoveryAgentStatus {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('Discovery agent status is success', () => {
    const { container, getByText } = render(<DiscoveryAgentStatus status="SUCCESS" />)

    expect(container).toHaveTextContent('SUCCESS')
    expect(getByText('SUCCESS')).toHaveClass('StyledProps--color-green700')
  })
  test('Discovery agent status is failed', () => {
    const { container, getByText } = render(<DiscoveryAgentStatus status="FAILED" />)

    expect(container).toHaveTextContent('FAILED')
    expect(getByText('FAILED')).toHaveClass('StyledProps--color-red700')
  })
  test('Discovery agent status is error', () => {
    const { container, getByText } = render(<DiscoveryAgentStatus status="ERROR" />)

    expect(container).toHaveTextContent('ERROR')
    expect(getByText('ERROR')).toHaveClass('StyledProps--color-orange700')
  })
  test('Discovery agent status is processed', () => {
    const { container, getByText } = render(<DiscoveryAgentStatus status="PROCESSED" />)

    expect(container).toHaveTextContent('PROCESSED')
    expect(getByText('PROCESSED')).toHaveClass('StyledProps--color-yellow700')
  })
})
