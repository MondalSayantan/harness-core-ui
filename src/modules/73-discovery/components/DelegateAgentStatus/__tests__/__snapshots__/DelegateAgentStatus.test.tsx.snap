// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discovery Agent Status  render 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--background-green50"
    style="padding: 2px 6px; border-radius: 4px;"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--color StyledProps--color-green700"
      style="font-weight: 600;"
    >
      SUCCESS
    </p>
  </div>
</div>
`;
